### DISCUSSION TITLE
Adding new nodes to existing cluster

### DISCUSSION THREAD

**Discussion by jagbarcelo** [techno-tim/k3s-ansible#24] [Questions] [answered]
Provided that you already have a running cluster after executing:

```
ansible-playbook site.yml -i inventory/my-cluster/hosts.ini
```

... what would be the right way to add several new hosts to it? Would it be ok just to edit `hosts.ini` to include the new names and re-run the former command again?

**Comment by NiclasPe** [comment, marked as answer]
I ask myself the same question, have you tried it?

**Reply by jagbarcelo** [reply]
Yes... adding new hosts to `hosts.ini`, inside their proper sections and replaying the `ansible-playbook` works fine.

It tries to do all the steps, but since most of the changes are already applied, they are skipped (no change) and where necessary does the changes, adding the new nodes without trashing the cluster.

At least it worked for me when I simply added a new regular worker `[node]`.

I didn't tested it for a `[manager]`, which would be good someone else confirms.

**Comment by J-Dre** [comment]
I've tested this for both a server and an agent node. it works fine for adding a new agent but fails to successfully add a new server node. In my case I was replacing one server node out of three.
It gets as far as the 'Verification' block of ther server set up but then fails so it appears that at leats one part of the previous set up tasks failed. They don't actually provide any errors through Ansible though.

If anyone knows if this _should_ work for replacing a server node that would be great to find out.

**Reply by J-Dre** [reply]
Update to this, it does work when using the IP address of the server in the ansible hosts file.

Between initially deploying the cluster and then running this again to replace a host I had changed my hosts file to use names for the servers and added 'host_vars' files for the IP addresses. That isn't compatible with this playbook.
